### ISSUE TITLE
Adjust position of significance levels from ggpubr in faceted boxplot

### ISSUE THREAD

**Issue by eggrandio** [kassambara/ggpubr#379]
Hello,

I am generating a faceted plot with different groups and I want to compare the mean of different groups to a control sample. I have a wide range of values, and want to position the "significance label" programmatically on top of each boxplot.

Here is some toy data.
```
library(ggpubr)
ggboxplot(mtcars, x = "am", y = "mpg") +
facet_wrap(~vs) +
stat_compare_means(aes(label = ..p.signif..),
                      method = "t.test", ref.group = "0")
```
Gives the following plot, that only takes into account the highest value in facet 1.
[![enter image description here][1]][1]

I would like to place the asterisk on the faceted plot 0 at the top of the box, like it is in the plot 1.
In this case I can extract the maximum value of each boxplot by:
```
lab_coords = mtcars %>% group_by(am,vs) %>% 
             summarize(max_mpg = max(mpg)) %>% pull(max_mpg)
lab_coords
[1] 19.2 24.4 26.0 33.9
```
When I pass the label coordinates to stat_compare_means I cannot place them in the correct order:
```
ggboxplot(mtcars, x = "am", y = "mpg") +
    facet_wrap(~vs) +
    stat_compare_means(aes(label = ..p.signif..),
                       method = "t.test", ref.group = "0",
                       label.y = lab_coords)
```
[![enter image description here][2]][2]

Is there any way of passing different ylabel positions to each facet?

  [1]: https://i.stack.imgur.com/yxoJb.png
  [2]: https://i.stack.imgur.com/Ii8dy.png

**Comment by kassambara** [comment]
yes, the current behavior of `stat_compare_means()` is to take the highest value in facet. I'm preparing an update with more and flexible features; should be available in the next release.


Here is a solution by using ggpubr and my (recent) rstatix package. (The next ggpubr release will heavily depends on the rstatix package to easily handle the stats parts)


``` r
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(rstatix))

# Data preparation ----------
# Make sure grouping variables are factors (not numeric)
df <- mtcars
df$am <- as.factor(df$am)
df$vs <- as.factor(df$vs)

# Statistical test------------
stat.test <- mtcars %>%
  group_by(vs) %>%
  wilcox_test(mpg ~ am, ref.group = "0") %>%
  add_significance()
stat.test 
#> # A tibble: 2 x 9
#>      vs .y.   group1 group2    n1    n2 statistic      p p.signif
#>   <dbl> <chr> <chr>  <chr>  <int> <int>     <dbl>  <dbl> <chr>   
#> 1     0 mpg   0      1         12     6        11 0.0216 *       
#> 2     1 mpg   0      1          7     7         5 0.0149 *

## Add stat test to a boxplot ---------
## Auto-add the xy position of p-values
stat.test <- stat.test %>%
  add_xy_position(x = "am")
# Boxplot with p-value
bxp <- ggboxplot(mtcars, x = "am", y = "mpg") +
  facet_wrap(~vs)  
bxp + stat_pvalue_manual(stat.test, tip.length = 0)
```

![](https://i.imgur.com/aBoPdQO.png)

``` r
# Remove the brackets
bxp + stat_pvalue_manual(
  stat.test, tip.length = 0,
  remove.bracket = TRUE
  )
```

![](https://i.imgur.com/aBCflX5.png)

[More examples: How to Add P-values to GGPLOT Facets](https://www.datanovia.com/en/blog/how-to-add-p-values-to-ggplot-facets/)

**Comment by pdeninis** [comment]
I made this plot some time ago.

![image](https://user-images.githubusercontent.com/35506062/112392584-edd77d80-8cf9-11eb-9227-3803abe73ad7.png)

As you can see, the brackets were perfectly positioned.

Today, after updating some packages - among which maybe ggpubr too - I have used the same code to redo the plot, and this is the result:

![image](https://user-images.githubusercontent.com/35506062/112392960-908ffc00-8cfa-11eb-84d8-bee969e8f006.png)

Have you some idea about what happened?

Thanks.

**Comment by pdeninis** [comment]
Continuing the previous comment...

It seems to me that the position computed for the first group has been used for the bracket of the third one, the position for the second has been used for the first group and that of the third group ended up below the second one...

**Comment by SplitInf** [comment]
Hi, is there any update to this feature? This is particularly useful for adding stats label with facet.

**Comment by kassambara** [comment]
This is supported now — `geom_pwc()` (added in 0.5.0) computes and places pairwise comparisons **per facet panel** automatically, so positions are relative to each panel rather than a global max:

``` r
suppressWarnings(suppressPackageStartupMessages(library(ggpubr)))
df <- ToothGrowth; df$dose <- factor(df$dose)
# geom_pwc() computes & places pairwise comparisons PER facet panel (ggpubr >= 0.5.0)
ggboxplot(df, x = "supp", y = "len", facet.by = "dose") +
  geom_pwc(aes(group = supp), method = "wilcox_test",
           label = "p.adj.signif", p.adjust.method = "BH")
```

![](https://i.imgur.com/qQAtwSC.png)

The manual alternative is `rstatix::add_xy_position()` + `stat_pvalue_manual()`.

Closing as resolved — reopen if anything's unclear. 🙌
